mantidproject/mantid
# Fix Mantid crashing if the ALC interface was closed while loading a data set.

**Description of work.**

This PR fixes an issue where mantid would crash if the ALC interface was closed while loading a set of data. The fix overwrites the closeEvent within ALCInterface to first cancel the loading algorithm. This allows the window to be closed.

Note:
Within Mantidplot, the event->ignore() causes an additional dialogue box to appear, which is slightly annoying as it leads to two dialogue boxes appearing. In the future it may be nice to create a single more specific dialogue box. For the mean time this solution is sufficient as it prevents Mantid crashing.

**To test:**
Open an ALC interface: Muon -> ALC
Enter some e.g (66666) and press auto
Click Load and then try to close interface
An error will been shown, which leads to the load being cancelled
The window should now be safe to close


Fixes #27231 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
